deploy/deployment-arm.yaml

```diff
@@ -21,7 +21,7 @@ spec:
       serviceAccountName: nfs-client-provisioner
       containers:
         - name: nfs-client-provisioner
-          image: quay.io/external_storage/nfs-client-provisioner-arm:latest
+          image: gcr.io/k8s-staging-sig-storage/nfs-subdir-external-provisioner:v4.0.0-rc2
```

Comment:
do we still need `deployment-arm.yaml` since the multi-arch image?

Reply:
Was planning on raising a clean-up PR with some more readme updates. But makes sense to include that in this PR as well. Could you please give the entire README a review as well please.